esmvaltool/recipes/recipe_ecs_constraints.yml

```diff
@@ -0,0 +1,1131 @@
+          - {dataset: HadGEM2-ES,    exp: abrupt4xCO2, start_year: 1860, end_year: 2009}
+          - {dataset: inmcm4,        exp: piControl,   start_year: 2090, end_year: 2239}
+          - {dataset: inmcm4,        exp: abrupt4xCO2, start_year: 2090, end_year: 2239}
+          # Only 140 years available
+          - {dataset: IPSL-CM5A-MR,  exp: piControl,   start_year: 1850, end_year: 1989}
+          - {dataset: IPSL-CM5A-MR,  exp: abrupt4xCO2, start_year: 1850, end_year: 1989}
+          - {dataset: IPSL-CM5B-LR,  exp: piControl,   start_year: 1850, end_year: 1999}
+          - {dataset: IPSL-CM5B-LR,  exp: abrupt4xCO2, start_year: 1850, end_year: 1999}
+          - {dataset: MIROC5,        exp: piControl,   start_year: 2100, end_year: 2249}
+          - {dataset: MIROC5,        exp: abrupt4xCO2, start_year: 2100, end_year: 2249}
+          - {dataset: MIROC-ESM,     exp: piControl,   start_year: 1880, end_year: 2029}
+          - {dataset: MIROC-ESM,     exp: abrupt4xCO2, start_year:    1, end_year:  150}
+          - {dataset: MPI-ESM-LR,    exp: piControl,   start_year: 1880, end_year: 2029}
+          - {dataset: MPI-ESM-LR,    exp: abrupt4xCO2, start_year: 1850, end_year: 1999}
+          - {dataset: MPI-ESM-MR,    exp: piControl,   start_year: 1850, end_year: 1999}
+          - {dataset: MPI-ESM-MR,    exp: abrupt4xCO2, start_year: 1850, end_year: 1999}
+          - {dataset: MPI-ESM-P,     exp: piControl,   start_year: 1866, end_year: 2015}
+          - {dataset: MPI-ESM-P,     exp: abrupt4xCO2, start_year: 1850, end_year: 1999}
+          - {dataset: MRI-CGCM3,     exp: piControl,   start_year: 1891, end_year: 2040}
+          - {dataset: MRI-CGCM3,     exp: abrupt4xCO2, start_year: 1851, end_year: 2000}
+          - {dataset: NorESM1-M,     exp: piControl,   start_year:  700, end_year:  849}
+          - {dataset: NorESM1-M,     exp: abrupt4xCO2, start_year:    1, end_year:  150}
+      tas_rtmt:
+        <<: *tas_settings_cmip5
+        additional_datasets: &rtmt_datasets_cmip5
+          - {dataset: IPSL-CM5A-LR,  exp: piControl,   start_year: 1850, end_year: 1999}
+          - {dataset: IPSL-CM5A-LR,  exp: abrupt4xCO2, start_year: 1850, end_year: 1999}
+      rtnt:
+        <<: *tas_settings_cmip5
+        short_name: rtnt
+        derive: true
+        additional_datasets: *rtnt_datasets_cmip5
+      rtmt:
+        <<: *tas_settings_cmip5
+        short_name: rtmt
+        additional_datasets: *rtmt_datasets_cmip5
+    scripts:
+      ecs:
+        script: climate_metrics/ecs.py
+        calculate_mmm: false
+        output_attributes:
+          var_type: label
+          tag: ECS
+          plot_ylim: [1.5, 6.0]
+          project: CMIP5
+          provenance_authors: ['schlund_manuel']
+          provenance_domains: ['global']
+          provenance_realms: ['atmos']
+          provenance_references: ['gregory04grl']
+          provenance_statistics: ['mean', 'anomaly']
+          provenance_themes: ['phys']
```

Comment:
`themes` and `realms` is usually included directly under the diagnostic in the recipe, have a look at how it's done in e.g. recipe_perfmetrics_CMIP5.yml or examples/recipe_python.yml. The other attributes are usually set in the code, because they are very closely tied to the diagnostic script.

Reply:
For the diagnostic [univariate_constraint.py](https://github.com/ESMValGroup/ESMValTool/blob/add_ecs_constraints/esmvaltool/diag_scripts/emergent_constraints/univariate_constraint.py) it does not make sense to hardcode the provenance information in the header of the recipe or the diagnostic script itself, since it is able to handle arbitrary emergent constraints. E.g., `realms` could be anything (atmosphere, ocean, land, etc.).

Reply:
But you will still be the author of the diagnostic script, right?